### PR TITLE
Update protobuf to 25.3

### PR DESCRIPTION
**Pull request by EbonJaeger**
**Summary**

- **gtest: Update to 1.14.0**
- **abseil-cpp: Build test helpers**
- **protobuf: Update to 25.3**
- **highway: Safety rebuild for gtest**
- **protobuf-c: Update to 1.5.0**
- **python-protobuf: Update to 25.3**
- **android-tools: Safety rebuild for protobuf**
- **astroid: Safety rebuild for protobuf**
- **cockatrice: Safety rebuild for protobuf**
- **libarcus: Update to 5.3.0**
- **range-v3: Update to 0.12.0**
- **curaengine: Update to 5.4.0**
- **libsavitar: Update to 5.3.0**
- **fdm-materials: Update to 5.4.0**
- **python-pyclipper: Add at 1.2.1**
- **python-uranium: Update to 5.6.0**
- **cura: Update to 5.4.0**
- **eternal-terminal: Safety rebuild for protobuf**
- **grpc: Update to 1.62.0**
- **protonmail-bridge: Safety rebuild for protobuf and grpc**
- **libcompizconfig: Safety rebuild for protobuf**
- **marble: Safety rebuild for protobuf**
- **mixxx: Safety update for protobuf**
- **mosh: Safety rebuild for protobuf**
- **mozc: Update to 2.29.5150.102**
- **mumble: Safety rebuild for protobuf**
- **qgis: Safety rebuild for protobuf**
- **paraview: Safety rebuild for protobuf**
- **ricochet: Safety rebuild for protobuf**
- **strawberry: Safety rebuild for protobuf**
- **telegram: Safety rebuild for protobuf**
- **usbguard: Safety rebuild for protobuf**
- **vlc: Safety rebuild for protobuf**
- **pytorch: Safety rebuild for protobuf**

**Test Plan**

Rebuilt all known packages against protobuf successfully. Launch Cura and go through the setup until the printer setup page.

**Checklist**

- [x] Package was built and tested against unstable

### CODE REVIEW

File: packages/l/libarcus/package.yml [hunk 1]
```diff
@@ -1,22 +1,52 @@
...
+    - https://github.com/Ultimaker/libArcus/archive/refs/tags/5.3.0.tar.gz : 3cf7411b573eda2de2aec9761bd6b539a5d5d67739761357f5f773d1490db0fb
+      # Use the git repo because the tarballs share the same name, and solbuild don't like that
+    - git|https://github.com/Ultimaker/pyArcus.git : 10e23195a57fca9807f6a5a7bb075c865de60059
 homepage   : https://github.com/Ultimaker/libArcus
-license    : LGPL-3.0-or-later
-component  : programming.library
-summary    : Communication library between internal components for Ultimaker software
+license    :
+    - LGPL-3.0-or-later
+      # The CMake files are licensed BSD
```
Comment (ReillyBrogan) [inline packages/l/libarcus/package.yml:11]: Yes but which one

File: packages/m/mozc/package.yml [hunk 2]
```diff
@@ -29,38 +35,41 @@ rundeps    :
...
     pushd ..
-    tar xf $sources/fcitx-mozc-icon.tar.gz
+    unzip $sources/ken_all.zip
+    unzip $sources/jigyosyo.zip
     popd
 
-    pushd src/third_party/gyp
-    git cherry-pick -n bc83cdacf5428ab6ddcc92fd0b0fc494cc9a4d4f
+    # Seed zip code dictionary generator
+    pushd src
```
Comment (ReillyBrogan) [inline packages/m/mozc/package.yml:47]: Maybe just one `pushd src/data/dictionary_oss` and `popd`?

File: packages/m/mozc/package.yml [hunk 3]
```diff
@@ -29,38 +35,41 @@ rundeps    :
...
+    pushd data
+    pushd dictionary_oss
+    PYTHONPATH=${PYTHONPATH}:../.. python3 ../../dictionary/gen_zip_code_seed.py --zip_code=../../../../KEN_ALL.CSV --jigyosyo=../../../../JIGYOSYO.CSV >> dictionary09.txt;
```
Comment (ReillyBrogan) [inline packages/m/mozc/package.yml:50]: Yes.

File: packages/p/paraview/package.yml [hunk 4]
```diff
@@ -44,6 +44,8 @@ builddeps  :
 libsplit   : no
 setup      : |
     %cmake_ninja \
+        -DCMAKE_CXX_STANDARD=17 \
+        -DVTK_IGNORE_CMAKE_CXX11_CHECKS=ON \
```
Comment (ReillyBrogan) [inline packages/p/paraview/package.yml:48]: Can you bump this second option down a bit so it's vaguely alphabeticalish

File: packages/p/protobuf-c/abi_used_libs [hunk 5]
```diff
@@ -1,7 +1,15 @@
 UNKNOWN
-ld-linux-x86-64.so.2
+libabsl_base.so.2308.0.0
+libabsl_die_if_null.so.2308.0.0
```
Comment (ReillyBrogan) [inline packages/p/protobuf-c/abi_used_libs:3]: Best library name

File: packages/r/ricochet/build.log
Comment (ReillyBrogan) [inline packages/r/ricochet/build.log:1]: Unorthodox file, but I'll take it

**Review by ReillyBrogan — CHANGES_REQUESTED** [review]
Some builddep/rundep ordering issues as well. Run `go-task check` locally you'll get the feedback sooner.

Also small nit, usually we prefer `%cmake_ninja` over `%cmake`, did you try that at all with some of the packages? It's fine if not why rock the boat on the Cura packages

File: packages/r/ricochet/build.log
Comment (EbonJaeger) [reply packages/r/ricochet/build.log:1]: Oops. Thought I got rid of that first.

File: packages/p/protobuf-c/abi_used_libs (hunk 5 above)
Comment (EbonJaeger) [reply packages/p/protobuf-c/abi_used_libs:3]: xD

File: packages/m/mozc/package.yml (hunk 2 above)
Comment (EbonJaeger) [reply packages/m/mozc/package.yml:47]: I didn't know you could do that. That's way better!

**Comment by EbonJaeger** [comment]
I just discovered that the shasums (and thus, presumably, the contents) of the additional zip code and jiqyosyo sources in `mozc` change periodically. In hindsight, I probably should have seen that coming. So, removing all that.

File: packages/c/cura/package.yml [hunk 6]
```diff
@@ -11,44 +12,86 @@ description: |
...
         -DCURA_CLOUD_API_ROOT="https://api.ultimaker.com" \
         -DCURA_CLOUD_ACCOUNT_API_ROOT="https://account.ultimaker.com" \
-        -DCURA_SDK_VERSION=6.0.0
+        -DLIB_SUFFIX:STR=
 build      : |
     %ninja_build
 install    : |
     %ninja_install
 
-    # for better performances
```
[the remark this answers is not in the document]
Comment (EbonJaeger) [reply packages/c/cura/package.yml:50]: The QML fails to compile:

```
Error compiling qml file: ./usr/share/cura/resources/qml/ToolTip.qml:1:1: error: Expected token `numeric literal'
```

It looks like a [blank file](https://github.com/Ultimaker/Cura/blob/main/resources/qml/ToolTip.qml)? It also hasn't changed in two years.

[the remark this answers is not in the document]
Comment (EbonJaeger) [reply packages/c/cura/package.yml:50]: It appears that it was moved into Uranium, and the blank file is the result of a merge conflict.

[the remark this answers is not in the document]
Comment (EbonJaeger) [reply packages/c/cura/package.yml:50]: Nuking that file seems to make things happy.

**Comment by EbonJaeger** [comment]
I didn't reorder the PyTorch dependency lists because they're broken up by what part of the program they're for; some are marked for ROCm.

**Review by ReillyBrogan — APPROVED** [review]
LGTM, Thanks!

**Pull request merged**